### PR TITLE
Added StepFunctionSyncExecutionPolicy to support StartSyncExecution for SFN express workflows.

### PR DESCRIPTION
**Pull request by avladi**
*Issue #, if available:*

*Description of changes:* Added "StepFunctionsSyncExecutionPolicy" policy to add support for synchronous execution of SFN express workflows.

*Description of how you validated changes:*
The policy is based on existing one for the StepFunction. The only difference is Action StartSyncExecution. I triple checked file formats.

*Checklist:*

- [x] Write/update tests
- [ ] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected

*Examples?*
```yaml
  StartSyncSFNExpressFunction:
    Type: AWS::Serverless::Function
    Properties:
      FunctionName: StartSyncSFNExpressFunctionName
      CodeUri: ../Lambdas/StartSyncSFNExpressFunction
      Handler: StartSyncSFNExpressFunction.StartExecution
      Runtime: python3.8
      Policies:
        - StepFunctionsSyncExecutionPolicy:
            StateMachineName: !GetAtt TestStepFunction.Name
```

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by avladi** [comment]
I checked failed build, I do not see anything wrong. I updated all documents where existing StepFunction policy is defined. I will try to run the failed test locally tomorrow.

**Comment by avladi** [comment]
The build is fixed, waiting for the review

**Review by hawflau — APPROVED** [review]
The change makes sense to me. We will request a security review internally as the change involves IAM policy changes.

File: samtranslator/policy_templates_data/policy_templates.json
Comment (hawflau) [inline samtranslator/policy_templates_data/policy_templates.json:2364]: Can we have a better name that mentions it is for Synchronous Express Workflows? It is also using the same name as in PR #1622 which covers the [standard synchronous workflow](https://docs.aws.amazon.com/step-functions/latest/dg/stepfunctions-iam.html#synchronous).

**Review by hawflau — CHANGES_REQUESTED** [review]

**Comment by hawflau** [comment]
Closing due to inactivity. Please open a new PR with the suggested change when it's ready. Thanks

**Pull request closed without merging**
